src/Service/Parsers/ODataASTVisitor.cs

```diff
@@ -251,5 +257,34 @@ private static string GetFilterPredicateOperator(UnaryOperatorKind op)
                     throw new ArgumentException($"Uknown Predicate Operation of {op}");
             }
         }
+
+        /// <summary>
+        /// Helper method to populate the DbType for the property referenced in the OData filter.
+        /// Since the nodes are processed in a postorder fashion, the current BinaryOperatorNode has already been processed,
+        /// and the parameter is already created for the property.
+        /// We just need to populate the DbType.
+        /// </summary>
+        /// <param name="nodeIn">Binary operator node<</param>
+        private void PopulateDbTypeForPropery(BinaryOperatorNode nodeIn)
+        {
+            SingleValuePropertyAccessNode propertyNode = nodeIn.Left.GetType() == typeof(SingleValuePropertyAccessNode) ?
+                    (SingleValuePropertyAccessNode)nodeIn.Left : (SingleValuePropertyAccessNode)nodeIn.Right;
+            string? paramName = $"{BaseQueryStructure.PARAM_NAME_PREFIX}param{_struct.Counter.Current() - 1}";
```

Comment:
how can we guarantee this would be the paramName?

Can we move this into a common function so that we can get the same paramName no matter where it is called from?

Reply:
The method summary illustrates why this would be the paramName and why it would always exist. Also, the current implementation is the fine and we can't move it into a common function.

Reply:
Let me clarify: 

How can you guarantee the format of the name of the parameter is going to be "@param{counter -1}"? 

You can easily right a static function:
`string MakeParamName(int counter)` to keep this in a centralized reusable location. 
Think if in future we decide to name the param as parameter - we need to only change it in 1 place instead of the 2 places you have right now: 
in this function and in [https://github.com/Azure/data-api-builder/blob/e810143c6bbe58d056181dda32b3703ea2ebb8ad/src/Service/Resolvers/BaseQueryStructure.cs#LL114C11-L114C11](https://github.com/Azure/data-api-builder/blob/e810143c6bbe58d056181dda32b3703ea2ebb8ad/src/Service/Resolvers/BaseQueryStructure.cs#LL114C11-L114C11)

Reply:
Gotcha, misunderstood what you meant.